eth_account/signers/ledger.py

```diff
@@ -0,0 +1,399 @@
+import struct
+import itertools
+
+import eth_utils
+import rlp
+
+from hexbytes import HexBytes
+
+from eth_account import (
+    Account,
+)
+
+from eth_account.datastructures import (
+    AttributeDict,
+)
+
+from eth_account.signers.base import (
+    BaseAccount,
+)
+
+from eth_account.internal.transactions import (
+    encode_transaction,
+    serializable_unsigned_transaction_from_dict,
+)
+
+from eth_utils import (
+    to_bytes,
+    to_hex,
+)
+
+from eth_utils.curried import (
+    keccak,
+)
+
+
+ETH_DERIVATION_PATH_PREFIX = "m/44'/60'/0'/"
+
+
+import hid
+import time
+class LedgerUsbDevice:
+    '''
+    References:
+    - https://github.com/LedgerHQ/blue-loader-python/blob/master/ledgerblue/comm.py#L56
+    - https://github.com/ethereum/go-ethereum/blob/master/accounts/usbwallet/ledger.go
+    '''
+
+    def __init__(self, debug=False):
+        hidDevicePath = None
+        for hidDevice in hid.enumerate(0, 0):
+            if hidDevice['vendor_id'] == 0x2c97:
+                if ('interface_number' in hidDevice and hidDevice['interface_number'] == 0) or ('usage_page' in hidDevice and hidDevice['usage_page'] == 0xffa0):
+                    hidDevicePath = hidDevice['path']
+        if hidDevicePath is not None:
+            dev = hid.device()
+            dev.open_path(hidDevicePath)
+            dev.set_nonblocking(True)
+        else:
+            raise Exception("No dongle found")
+        self.device = dev
+        self.debug = debug
+
+    def exchange(self, apdu, timeout=20000):
+        if self.debug:
+            print("HID => %s" % to_hex(apdu))
+
+        # Construct the wraped payload
+        apdu = self.wrapCommandAPDU(0x0101, apdu, 64)
+        padSize = len(apdu) % 64
+        tmp = apdu
+        if padSize != 0:
+            tmp.extend([0] * (64 - padSize))
+
+        # Send to device
+        offset = 0
+        while(offset != len(tmp)):
+            data = tmp[offset:offset + 64]
+            data = bytearray([0]) + data
+            if self.device.write(data) < 0:
+                raise BaseException("Error while writing")
+            offset += 64
+
+        # Receive reply
+        dataLength = 0
+        dataStart = 2
+        result = self.waitFirstResponse(timeout)
+        self.device.set_nonblocking(False)
+        while True:
+            response = self.unwrapResponseAPDU(0x0101, result, 64)
+            if response is not None:
+                result = response
+                dataStart = 0
+                swOffset = len(response) - 2
+                dataLength = len(response) - 2
+                self.device.set_nonblocking(True)
+                break
+            result.extend(bytearray(self.device.read(65)))
+        sw = (result[swOffset] << 8) + result[swOffset + 1]
+        response = result[dataStart : dataLength + dataStart]
+        if self.debug:
```

Comment:
Can you drop the `debug` flag and instead use the standard python logging facilities.  We don't ship  with `print` statements.

```
class LedgerUsbDevice:
    logger = logging.getLogger('eth_account.signers.ledger.LedgerUsbDevices')

    def some_method(self):
        self.logging.debug('the message')
```